Mathlib/RingTheory/PowerSeries/Ideal.lean

```diff
@@ -0,0 +1,218 @@
+      sub_self, map_zero, sub_zero, coeff_C_mul] at h
+    refine (I_prime.mul_mem_iff_mem_or_mem.1 (h ▸ I.sub_mem (G hI n).2 ?_)).resolve_left hI
+    exact  I.sum_mem fun i _ ↦ I.mul_mem_left _ (F_apply_mem_I haI i)⟩
+
+private lemma G_sub_sum_T (n : ℕ) (hI : X ∉ I) :
+    (G hg haI hI n).1 - ∑ i, C (T (G hg haI hI n).2 haI i) * F haI i =
+      X * (G hg haI hI (n + 1)).1 := by
+  simpa [sum_T, constantCoeff_F_apply] using sub_const_eq_X_mul_shift
+    ((G hg haI hI n).1 - ∑ i, C (T (G hg haI hI n).2 haI i) * F haI i)
+
+/-- Given `a : Fin k → R` such that `(haP : span (range a) = I.map constantCoeff)` and `g : R⟦X⟧`
+such `(hg : g ∈ I)`, we construct `H : Fin k → R⟦X⟧` such that `g = ∑ i, H i * F i`. -/
+private noncomputable def H (hI : X ∉ I) (i : Fin k) : R⟦X⟧ :=
+  mk fun n ↦ T (G hg haI hI n).2 haI i
+
+private lemma g_sub_sum_trunc_H (hI : X ∉ I) (n : ℕ) :
+    g - ∑ i, trunc n (H hg haI hI i) * F haI i = X ^ n * (G hg haI hI n).1 := by
+  induction n with
+  | zero => simp [G]
+  | succ n hn =>
+    conv =>
+      enter [1, 2, 2, i]
+      rw [trunc_succ, Polynomial.coe_add, Polynomial.coe_monomial, add_mul, ← mul_one (coeff n _),
+        ← smul_eq_mul (b := 1), map_smul, ← X_pow_eq, smul_eq_C_mul, mul_comm _ (X ^ n), mul_assoc]
+    rw [sum_add_distrib, sub_add_eq_sub_sub, hn, ← mul_sum, ← mul_sub]
+    simp only [H, coeff_mk, G_sub_sum_T]
+    ring
+
+private lemma sum_H_mul_F (hI : X ∉ I) : ∑ i, H hg haI hI i * F haI i = g := by
+  symm
+  refine sub_eq_zero.1 ?_
+  ext n
+  conv =>
+    enter [1, 2, 2, 2, i]
+    rw [eq_X_pow_mul_shift_add_trunc (n + 1) (H hg haI hI i), add_comm, add_mul, mul_assoc]
+  rw [sum_add_distrib, sub_add_eq_sub_sub, g_sub_sum_trunc_H, ← mul_sum]
+  simp [coeff_X_pow_mul']
+
+private theorem I_eq_span_range (hI : X ∉ I) : I = span (range (F haI)) :=
+  le_antisymm
+    (fun _ hg ↦ mem_span_range_iff_exists_fun.2 ⟨_, sum_H_mul_F hg haI hI⟩)
+    (span_le.2 <| range_subset_iff.2 <| F_apply_mem_I haI)
+
+/-- Given a prime ideal `I` of `R⟦X⟧` such that `X ∉ I`, if `I.map constantCoeff` is generated by a
+finite set `S : Set T`, then there exists `T : Set R`, of the same cardinality as `S`, that
+generates `I`. -/
+theorem exist_eq_span_eq_ncard_of_X_not_mem (hI : X ∉ I) {S : Set R}
+    (hSI : span S = I.map constantCoeff) (hS : S.Finite) :
+      ∃ T, I = span T ∧ T.Finite ∧ T.ncard = S.ncard := by
+  obtain ⟨k, a, a_injective, rfl⟩ := Finite.fin_param hS
+  refine ⟨_, I_eq_span_range hSI hI, finite_range _, trans (b := k) ?_ ?_⟩ <;>
+  rw [Set.ncard_range_of_injective, Nat.card_fin]
+  · intro _ _ this
+    apply_fun constantCoeff at this
+    simp only [constantCoeff_F_apply] at this
+    exact a_injective this
+  · exact a_injective
+
+end X_notMem
+
+/-- A prime ideal `P` of `R⟦X⟧` is finitely generated if and only if `P.map constantCoeff` is
+finitely generated. -/
+lemma IsPrime.FG_iff {P : Ideal R⟦X⟧} [P.IsPrime] : P.FG ↔ (P.map constantCoeff).FG := by
```

Comment:
```suggestion
lemma IsPrime.fg_iff {P : Ideal R⟦X⟧} [P.IsPrime] : P.FG ↔ (P.map constantCoeff).FG := by
```
According to the naming convention, `FG` should be in camel case (i.e. first letter lowercase) and acronyms get the same case as its first letter so it's `fg`.